### PR TITLE
feat: add simple address validity check

### PR DESCRIPTION
**Pull request by leohhhn**
## Description

This PR adds a simple `Valid()` check to the Address type. It is there to prevent most human errors, and doesn't actually validate the math behind bech32 (for now) - it relies only on the length of the address. 

Running `gno test .` will run the unit test for the `Valid()` function.
Closes: #1298 

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [x] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

**Review by thehowl — APPROVED** [review]

**Review by harry-hov — APPROVED** [review]
LGTM 💯 

I just wonder if we can get rid of these in the same PR:
 https://github.com/gnolang/gno/blob/34d78b3f9d4afb51d3f53d2d7d73752bb59ff4ff/examples/gno.land/p/demo/grc/grc20/util.gno#L7-L12

https://github.com/gnolang/gno/blob/34d78b3f9d4afb51d3f53d2d7d73752bb59ff4ff/examples/gno.land/p/demo/grc/grc721/util.gno#L9-L14

https://github.com/gnolang/gno/blob/34d78b3f9d4afb51d3f53d2d7d73752bb59ff4ff/examples/gno.land/p/demo/grc/grc1155/util.gno#L9-L14

File: gnovm/stdlibs/std/crypto.gno [hunk 1]
```diff
@@ -6,6 +6,11 @@ func (a Address) String() string {
 	return string(a)
 }
 
+// Valid checks if the address is of specific length. Doesn't check prefix or checksum for the address
+func (a Address) Valid() bool {
```
[the remark this answers is not in the document]
Comment (zivkovicmilos) [reply gnovm/stdlibs/std/crypto.gno]: Resolved by [0d5a31b](https://github.com/gnolang/gno/pull/1303/commits/0d5a31bdc0739bbac59380878a99666839435bd3)

**Review by moul — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -3,9 +3,9 @@ module github.com/gnolang/gno
 go 1.20
 
 require (
-	github.com/btcsuite/btcd v0.22.0-beta.0.20220111032746-97732e52810c
+	github.com/btcsuite/btcd v0.22.3
 	github.com/btcsuite/btcd/btcutil v1.0.0
-	github.com/btcsuite/btcutil v1.0.2
+	github.com/btcsuite/btcutil v1.0.3-0.20201208143702-a53e38424cce
```
Comment (thehowl) [inline go.mod:8]: Ehh, I'd say upgrading btcd/btcutil is something that should be done in another PR, especially considering https://github.com/gnolang/gno/pull/1064

I'd suggest reverting changes here + go.sum

Comment (leohhhn) [reply go.mod:8]: It was accidental, sorry. Will revert them :)

Comment (moul) [reply go.mod:8]: Oops, my mistake. I attempted to fix the previous commit but didn't thoroughly check it.
